### PR TITLE
nasin-nanpa: init at 2.5.1

### PR DESCRIPTION
**Pull request by somasis**
###### Description of changes

A font for sitelen pona, a hieroglyphic writing system for the constructed language Toki Pona.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/fonts/nasin-nanpa/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    mkdir -p $out/share/fonts/opentype
+    for f in $srcs; do
+        n=$(stripHash $f)
+        cp $f $out/share/fonts/opentype/$n
+    done
+  '';
+
+  outputHashAlgo = "sha256";
+  outputHashMode = "recursive";
+
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/nasin-nanpa/default.nix:32]:
```suggestion
```
This is unused

File: pkgs/data/fonts/nasin-nanpa/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  srcs = [
+    (fetchurl {
+      name = "${pname}.otf";
+      url = "https://github.com/ETBCOR/${pname}/releases/download/n${version}/${pname}-${version}.otf";
+      hash = "sha256-++uOrqFzQ6CB/OPEmBivpjMfAtFk3PSsCNpFBjOtGEg=";
+    })
+    (fetchurl {
+      name = "${pname}-lasina-kin.otf";
+      url = "https://github.com/ETBCOR/${pname}/releases/download/n${version}/${pname}-${version}-lasina-kin.otf";
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/nasin-nanpa/default.nix:15]:
```suggestion
      name = "nasin-nanpa.otf";
      url = "https://github.com/ETBCOR/nasin-nanpa/releases/download/n${version}/nasin-nanpa-${version}.otf";
      hash = "sha256-++uOrqFzQ6CB/OPEmBivpjMfAtFk3PSsCNpFBjOtGEg=";
    })
    (fetchurl {
      name = "nasin-nanpa-lasina-kin.otf";
      url = "https://github.com/ETBCOR/nasin-nanpa/releases/download/n${version}/nasin-nanpa-${version}-lasina-kin.otf";
```
Substituting pname wherever possible just makes the code harder to read

File: pkgs/data/fonts/nasin-nanpa/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    })
+  ];
+
+  dontUnpack = true;
+
+  installPhase = ''
+    mkdir -p $out/share/fonts/opentype
+    for f in $srcs; do
+        n=$(stripHash $f)
+        cp $f $out/share/fonts/opentype/$n
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/nasin-nanpa/default.nix:26]:
```suggestion
    for src in $srcs; do
      file=$(stripHash $src)
      cp $src $out/share/fonts/opentype/$file
```

File: pkgs/data/fonts/nasin-nanpa/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    done
+  '';
+
+  outputHashAlgo = "sha256";
+  outputHashMode = "recursive";
+
+  meta = with lib; {
+    homepage = "https://github.com/ETBCOR/${pname}";
+    description =
+      "UCSUR OpenType monospaced font for the Toki Pona writing system, Sitelen Pona";
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/nasin-nanpa/default.nix:36]:
```suggestion
    description = "UCSUR OpenType monospaced font for the Toki Pona writing system, Sitelen Pona";
```

File: pkgs/data/fonts/nasin-nanpa/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    longDescription = ''
+      ni li nasin pi sitelen pona.
+      sitelen ale pi nasin ni li sama mute weka.
+      sitelen pi nasin ni li lon nasin UCSUR kin.
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/nasin-nanpa/default.nix:40]: This needs an english translation

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/data/fonts/nasin-nanpa/default.nix (hunk 5 above)
Comment (somasis) [reply pkgs/data/fonts/nasin-nanpa/default.nix:40]: It is a translation of the short description to toki pona.

Back-translating, it reads
> This is a font/style/system for writing sitelen pona.
> All the characters in this font are the same amount of space. [which is to say, they're monospaced]
> The symbols of this font are available via the [UCSUR](https://www.kreativekorp.com/ucsur/) system, too.

**Pull request merged**